src/libraries/System.Security.Cryptography.Cose/tests/CoseHeaderMapTests.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+using System.Formats.Cbor;
+using System.Collections.Generic;
+using static System.Security.Cryptography.Cose.Tests.CoseTestHelpers;
+
+namespace System.Security.Cryptography.Cose.Tests
+{
+
+    public class CoseHeaderMapTests
+    {
+        [Fact]
+        public void SetValue_GetValue_KnownHeader()
+        {
+            var alg = new CoseHeaderLabel(KnownHeaders.Alg);
+            var contentType = new CoseHeaderLabel(KnownHeaders.ContentType);
+            var kid = new CoseHeaderLabel(KnownHeaders.Kid);
+            var iV = new CoseHeaderLabel(KnownHeaders.IV);
+            var partialIV = new CoseHeaderLabel(KnownHeaders.PartialIV);
+
+            var map = new CoseHeaderMap();
+            map.SetValue(alg, (int)ECDsaAlgorithm.ES256);
+            map.SetValue(contentType, ContentTypeDummyValue);
+            map.SetValue(kid, s_SampleContent);
+            map.SetValue(iV, ReadOnlySpan<byte>.Empty);
+            map.SetValue(partialIV, ReadOnlySpan<byte>.Empty);
+
+            Assert.Equal((int)ECDsaAlgorithm.ES256, map.GetValueAsInt32(alg));
+            Assert.Equal(ContentTypeDummyValue, map.GetValueAsString(contentType));
+            Assert.True(map.GetValueAsBytes(kid).SequenceEqual(s_SampleContent));
+            Assert.True(map.GetValueAsBytes(iV).SequenceEqual(ReadOnlySpan<byte>.Empty));
+            Assert.True(map.GetValueAsBytes(partialIV).SequenceEqual(ReadOnlySpan<byte>.Empty));
+        }
+
+        [Theory]
+        [MemberData(nameof(KnownHeadersEncodedValues_TestData))]
+        public void SetEncodedValue_GetEncodedValue_KnownHeader(int knownHeader, ReadOnlyMemory<byte> encodedValue)
```

Comment:
Make the test fail locally (my favorite way, add `Assert.False(true, "Test Passed");` as the last line) and look at what xUnit reports as the failing test name.  Do you think that's identifiable enough?  If yes, no worries.  If no, consider changing how it's populated until you're happy with the test identifier.

Reply:
It looks pretty identifiable to me. However I changed the last bit of the name to be more specific:
~KnownHeader~ -> KnownCoseHeaderLabel

```
System.Security.Cryptography.Cose.Tests.CoseHeaderMapTests.SetEncodedValue_GetEncodedValue_KnownCoseHeaderLabel(knownHeader: 1, encodedValue: System.ReadOnlyMemory<Byte>[1]) [FAIL]
        Test Passed
        Expected: False
        Actual:   True
        Stack Trace:
          C:\repos\runtime\src\libraries\System.Security.Cryptography.Cose\tests\CoseHeaderMapTests.cs(44,0): at System.Security.Cryptography.Cose.Tests.CoseHeaderMapTests.SetEncodedValue_GetEncodedValue_KnownCoseHeaderLabel(Int32 knownHeader, ReadOnlyMemory`1 encodedValue)
```